File: imdb-front/src/app/interfaces/movieDTO.ts
import {CrewDTO} from "./crewDTO";
import {ReviewDTO} from "./reviewDTO";
import {VideoDTO} from "./videoDTO";
import {GenreDTO} from "./genreDTO";

export interface MovieDTO {

  id: number;

  title: string;

  imdbId: string;

  budget: number;

  overview: string;

  posterPath: string;

  upComing: boolean;

  releaseDate: string;

  originalLanguage: string;

  runtime: number;

  voteAverage: number;

  voteCount: number;

  popularity: number;

  genres: Set<GenreDTO>;

  crewSet: Set<CrewDTO>;

  reviews: Set<ReviewDTO>;

  video: VideoDTO;

}

File: imdb-back/src/main/java/com/ihu/imdbback/dto/MovieDTO.java
package com.ihu.imdbback.dto;

import com.ihu.imdbback.entity.Video;
import lombok.Getter;
import lombok.Setter;

import java.util.Set;

@Getter
@Setter
public class MovieDTO {

  private Integer id;

  private String title;

  private String imdbId;

  private Integer budget;

  private String overview;

  private String posterPath;

  private boolean upComing;

  private String releaseDate;

  private String originalLanguage;

  private Integer runtime;

  private Double voteAverage;

  private Integer voteCount;

  private Double popularity;

  private Set<GenreDTO> genres;

  private Set<CrewDTO> crewSet;

  private Set<ReviewDTO> reviews;

  private Video video;

}
